aws/request/request.go

```diff
@@ -77,13 +75,10 @@ func New(cfg aws.Config, clientInfo metadata.ClientInfo, handlers Handlers,
 		p = "/"
 	}
 
-	httpReq, _ := http.NewRequest(method, "", nil)
-
-	var err error
-	httpReq.URL, err = url.Parse(clientInfo.Endpoint + p)
+	httpReq, err := http.NewRequest(method, clientInfo.Endpoint+p, nil)
 	if err != nil {
-		httpReq.URL = &url.URL{}
 		err = awserr.New("InvalidEndpointURL", "invalid endpoint uri", err)
+		return &Request{Error: err}
```

Comment:
I'm not sure if this can exit early here. It is possible for user's custom request handlers to depend on the `request.Request.HTTPRequest` value expecting it to be set. Returning early without the other fields could cause breaking changes because fields like HTTPRequest are no longer set. Its still an error condition, but exiting early may cause applications to panic instead of returning a API operation error.

I don't see any problem with getting ride of the `url.Parse` and letting `NewRequest` validate the URL. Both will report the error in Go 1.6. Neither 1.4 or 1.5 would return an error for invalid URLs such as `http://example.com:80 `. For 1.4 and 1.5 the request would fail at transmission time instead of being caught early. #689.